tmva/tmva/inc/TMVA/DecisionTree.h

```diff
@@ -187,9 +192,35 @@ namespace TMVA {
       inline void SetUseExclusiveVars(Bool_t t=kTRUE){fUseExclusiveVars = t;}
       inline void SetNVars(Int_t n){fNvars = n;}
 
+      #ifdef R__USE_IMT
+      void InitThreadExecutor(UInt_t nthreads){
+          fPool.reset(new ROOT::TThreadExecutor(nthreads));
+          fNumCPUs = nthreads;
+      };
+      #endif
+
 
    private:
       // utility functions
+      
+      // Only include parallelizatiion if the multithreading compilation flag is turned on
+      #ifdef R__USE_IMT 
+      // number of CPUs available for parallelization
+      UInt_t fNumCPUs = 1;
+      UInt_t GetNumCPUs(){
+         SysInfo_t s;
+         gSystem->GetSysInfo(&s);
+         UInt_t ncpu  = s.fCpus;
+         return ncpu;
+      };
+
+      // #### root multithreading object
+      // #### had to define as pointer so that we can initialize the number of CPUs to use AFTER
+      // #### getting information from the user. Needed to do this to automate timing vs nCPUs during
+      // #### development.
+      std::unique_ptr<ROOT::TThreadExecutor> fPool = std::unique_ptr<ROOT::TThreadExecutor>(nullptr);
```

Comment:
I mainly created options for nthreads so that I could do the bench marking studies. The option is defaulted to use all of the available CPUs. If you would like the user to be able to set the number of CPUs then let them know about the option and their choice will override the default. If not, just don't let them know about it and it will be set automatically. The way this is coded is flexible, though I'm not sure how it differs from the default constructor.

Anyways, I would argue to default the threading option to the desired behavior for an uninformed user, but allow an informed user to have greater control.